### PR TITLE
Add support for layouts + sending email to anyone with shared secret

### PR DESCRIPTION
**Pull request by blambeau**

### CODE REVIEW

**Comment by blambeau** [comment]
@OSdave @ciremoussadia what do you think about those two new features?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,35 @@
-# 0.3.0 - 2021/03/26
+## 1.2.0
+
+* Add support for a TALKTOME_LAYOUTS_FOLDER environment variable used
+  in auto_options. When set, the folder is used to embed formatted emails
+  in more general layouts (see Client::Local.templater).
```
Comment (ciremoussadia) [inline CHANGELOG.md:5]: I'm afraid there may be confusion between layout and templates for people like me.

**Review by ciremoussadia — COMMENTED** [review]
Do you think after that we can improve the Dockerfile?

File: lib/talktome.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@ def with_env(which, &bl)
   def set_env(which, value, &bl)
     old, ENV[which] = ENV[which], value
     bl.call.tap{
-      ENV[which] = old unless old.nil?
+      ENV[which] = old
     }
   end
   module_function :set_env
```
Comment (OSdave) [inline lib/talktome.rb:32]: I don't understand this code, if one day you have time to explain it to me

**Review by OSdave — APPROVED** [review]

File: lib/talktome.rb (hunk 2 above)
Comment (blambeau) [reply lib/talktome.rb:32]: Have a look at `Object#tap`, that's what's missing in your mental picture. Basically, the method takes a block, saves an env var before the block, executes and returns the block result, but restores the env var afterwards. Same as

```
old = x
x = y
result = bl.call
x = old
return result

**Pull request merged**